feat: add triage-issues + triage-board skills (+ visual-test hardening) (#36012)
microsoft/fluentui: .agents/skills/visual-test/SKILL.md

> npm ls -g @playwright/cli 2>/dev/null || npm install -g @playwright/cli@0.1.1

Reviewer: this is not very user friendly on dev box as we enforce global creep in user land. we should either make the package part of our devDeps or enforce `npx`

Author: Going with `npx` over devDep, agreed on the global-creep concern. These skills are agent-invoked and opt-in, so adding a root devDep that every contributor pays for during yarn install does not pencil out for a tool only a small subset uses. Switched every invocation to `npx -y @playwright/cli@0.1.1 <command>` in `010ac9f881`, dropped the global install prereqs in both skills. Version pin stays inline, first call seeds the npx cache, no PATH pollution.

```

## Critical: use the per-component Storybook only

> **Never** use `yarn storybook` at the repo root, `public-docsite-v9`, or any workspace-wide Storybook build for visual validation. They pull in every v9 package (including deprecated `unstable` re-exports) and will either fail to compile due to missing packages or get stuck in HMR restart loops — the validation becomes useless. Always boot the **per-component stories package**, which only imports its own component's stories and dependencies.

Reviewer: ```suggestion
Always boot the **per-component stories package** (`react-<component>-stories`) via nx `storybook` target, which only imports its own component's stories and dependencies.
```

## Steps

1. **Find the component's stories package.** Each v9 component has a dedicated stories package named `react-<component>-stories`:

   ```bash

>    yarn nx show projects 2>/dev/null | grep "^react-<lowercase-component-name>-stories$"

Reviewer: ```suggestion
   yarn --silent nx show project react-<lowercase-component-name>-stories --json
```

   ```

>    If the grep returns nothing, the component doesn't have its own stories package — check for a preview package (`react-<component>-preview-stories`) or ask before proceeding.

Reviewer: ```suggestion
   If nx returns nothing with output of `Could not find project react-<component>-stories`, the component doesn't have its own stories package — check for a preview package (`react-<component>-preview-stories`) or ask before proceeding.
```

2. **Start the component's Storybook dev server.** Use the `storybook` target on the stories project directly — it's the most portable, since library aliases like `react-<component>:start` were only added in April 2026 and may not exist in older workspace snapshots:

   ```bash

>    yarn nx run react-<component>-stories:storybook --skip-nx-cache &

Reviewer: ```suggestion
   yarn nx run react-<component>-stories:storybook &
```

>    NX_PID=$!

Reviewer: why do we store this when it's not used anywhere ?

Author: You are right, it was dead code. The nx wrapper exits 0 shortly after delegating to storybook (the next paragraph in the skill even documents this), so the saved PID would point at a corpse anyway. Cleanup tracks the real storybook child via `SB_CHILD` from `pgrep`. Dropped the line in `3870966725`.

   ```

>    `--skip-nx-cache` matters: the `storybook` target is (unusually) declared with `cache: true`, and a prior run can replay cached output and exit without actually starting a server.

Reviewer: this is not true

```suggestion
```

   ```

5. **Navigate to the specific story iframe** and capture a screenshot.
   Use the iframe URL for a clean render without Storybook chrome:

   ```bash
   playwright-cli goto "http://localhost:$SB_PORT/iframe.html?id=components-<component>--default&viewMode=story"
   playwright-cli screenshot --filename=/tmp/visual-test-$ARGUMENTS.png
   ```

6. **View the screenshot** using the Read tool to visually inspect the rendered component.

7. **Use `snapshot`** to get the accessibility tree and find interactive element refs:

   ```bash
   playwright-cli snapshot
   ```

   Then interact with elements by ref (e.g., click, hover) before taking more screenshots.

8. **If the component doesn't look right**, go back to the code, fix the issue, and repeat from step 4 (Storybook hot-reloads changes).

9. **Clean up** when done:
   ```bash
   playwright-cli close
   # Kill storybook — the nx wrapper may already be gone, so target the child
   [ -n "$SB_CHILD" ] && kill "$SB_CHILD" 2>/dev/null
   lsof -i :$SB_PORT -t 2>/dev/null | xargs kill 2>/dev/null
   ```

## Troubleshooting

**`yarn nx run react-<component>-stories:storybook` says the target doesn't exist.**

> The workspace graph may be stale (recent reparent). Run `yarn nx reset` then retry. If `start` aliases still don't exist, use the direct yarn invocation:

Reviewer: why are we mixing storybook and start targets ?

```suggestion
The workspace graph may be stale (recent reparent). Run `yarn nx reset` then retry. If `stroybook` aliases still don't exist, use the direct yarn invocation:
```

```

**The build stops mid-compilation and nx "completes" with exit 0.**
That's the nx cache replaying a prior partial run. Always pass `--skip-nx-cache`.

**The build fails with `Module not found: @fluentui/react-alert`, `@fluentui/react-infobutton`, or `@fluentui/react-virtualizer`.**
These three packages are listed as deps of `@fluentui/react-components` and are linked into `node_modules` from local workspace sources (`packages/react-components/deprecated/react-alert/`, `deprecated/react-infobutton/`, `react-virtualizer/`). Their `lib-commonjs/` output only exists after they're built. On a fresh clone this build hasn't run yet, so the linked entry points at files that don't exist. Fix:

```bash
yarn nx run-many -t build -p react-alert,react-infobutton,react-virtualizer
```

Then restart the storybook target. This is a real infrastructure wart — it affects **any** per-component Storybook that pulls stories from the shared `@fluentui/react-components` barrel, not just combobox. (If you see the error elsewhere in this repo — e.g. the workspace-wide Storybook — the same fix applies plus the "don't use workspace-wide" rule above still stands.)

> 

Reviewer: not true

```suggestion
```

**The story loads but keeps reloading (`[HMR] Cannot find update (Full reload needed)`).**

> Same root cause — you're on the workspace-wide Storybook, which has HMR issues when preview packages rebuild. Kill it and use the per-component one.

Reviewer: where is this coming from ?

Author: Fair callout. This bullet was holdover from an earlier draft where the agent might fall back to the workspace-wide Storybook on missing-target errors. Now that the prereqs require the per-component stories package (and a separate troubleshooting bullet covers the only legit fallback, direct `yarn storybook dev` inside the stories package), a tip that says "kill the workspace-wide one" has no referent in any path the skill recommends. The "Same root cause" handwave also did not actually link back to the previous bullet, since stale workspace graph and HMR reload loops are different failure modes. Removed in `efe1c3f9bc`.
